server/grpc_bytestream.go

```diff
@@ -325,6 +325,11 @@ func (s *grpcServer) parseWriteResource(r string) (string, int64, casblob.Compre
 			status.Errorf(codes.InvalidArgument, "Invalid size (must be non-negative): %d from %q", size, r)
 	}
 
```

Comment:
I think we should perform a similar check in `BatchUpdateBlobs` and `UpdateActionResult`, and possibly also return a value other than `0` in `CacheCapabilities.MaxBatchTotalSizeBytes` if `max_blob_size` is a positive number less than 4M.

Reply:
I think this might be the last part missing before this can land.

Reply:
Acrtually, maybe we can change this a little- if we move the check into `disk.Put` then it would apply to all blobs the same way, regardless of whether they came from the HTTP or gRPC API. The errors might reach the client slightly later, but that doesn't need to be fast since it probably requires human intervention to fix anyway.

This won't affect loading preexisting blobs from the cache (unless we choose to add a check to `disk.loadExistingFiles` too).

Reply:
that seems like a substantially different PR? I'm not very familiar with the architecture here, but I can try putting a check in that spot and verify that has the behavior I need and send another PR.

I hope you're also considering whether it's economical to lead me through making this change, vs. just making it yourself? I have no preference.

Reply:
Sure, I can finish this off for you.

Reply:
That's awesome, thank you! I know it's hard to hand-hold contributors who don't know their way around the codebase, and even after that they don't become regular contributors :(

Reply:
No problem.

Here's the reworked version: https://github.com/buchgr/bazel-remote/pull/446